portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,76 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+function getdataservlet(){
+    console.log('Fetching data servlet');
+    const responsePromise = fetch('/data');
```

Comment:
What you have here should work, but if you want to simplify this code, you can replace your calls to handleReponse() and addDataServletToDom() with just three lines of code:  
fetch('/data').then(response => response.json()).then((quote) => {
    document.getElementById('dataservlet-container').innerText = quote;
  });